servant-server/src/Servant/Server/Internal.hs

```diff
@@ -398,11 +399,17 @@ instance HasServer Raw context where
   type ServerT Raw m = Application
 
   route Proxy _ rawApplication = RawRouter $ \ env request respond -> do
-    r <- runDelayed rawApplication env request
-    case r of
-      Route app   -> app request (respond . Route)
-      Fail a      -> respond $ Fail a
-      FailFatal e -> respond $ FailFatal e
+    -- note: a Raw application doesn't register any cleanup
```

Comment:
I *think* this is not just for consistency. E.g.:

```
type API = FileUpload :> Raw
```

Isn't in this case not having this going to cause the cleanup action to not be run? (It probably makes sense to have a test for (1) cleanup being run for `Raw`; (2) cleanup only being run *once*)